Source/GRChomboCore/BoundaryConditions.cpp

```diff
@@ -514,6 +667,100 @@ void BoundaryConditions::fill_reflective_cell(
+    const int dir, const std::vector<int> &extrapolating_comps,
+    const int order) const
+{
+    for (int icomp : extrapolating_comps)
+    {
+        // current radius
+        double radius = Coordinates<double>::get_radius(
+            iv, m_dx, {m_center[0], m_center[1], m_center[2]});
+
+        // vector of 2 nearest values and radii within the grid
+        std::array<double, 2> value_at_point;
+        std::array<double, 2> r_at_point;
+        // how many units are we from domain boundary?
+        int units_from_edge = 0;
+        if (a_side == Side::Hi)
+        {
+            // how many units are we from domain boundary?
+            units_from_edge = iv[dir] - m_domain_box.bigEnd(dir);
+            // vector of 2 nearest values and radii within the grid
+            for (int i = 0; i < 2; i++)
+            {
+                IntVect iv_tmp = iv;
+                iv_tmp[dir] += -units_from_edge - i;
+                FOR1(idir)
+                {
+                    if (iv_tmp[idir] > m_domain_box.bigEnd(idir))
+                    {
+                        iv_tmp[idir] = m_domain_box.bigEnd(idir);
+                    }
+                    else if (iv_tmp[idir] < m_domain_box.smallEnd(idir))
+                    {
+                        iv_tmp[idir] = m_domain_box.smallEnd(idir);
+                    }
+                }
+                value_at_point[i] = out_box(iv_tmp, icomp);
+                r_at_point[i] = Coordinates<double>::get_radius(
+                    iv_tmp, m_dx, {m_center[0], m_center[1], m_center[2]});
+            }
+        }
+        else // Lo side
+        {
+            // how many units are we from domain boundary?
+            units_from_edge = -iv[dir] + m_domain_box.smallEnd(dir);
+            // vector of 2 nearest values within the grid
+            for (int i = 0; i < 2; i++)
+            {
+                IntVect iv_tmp = iv;
+                iv_tmp[dir] += units_from_edge + i;
+                FOR1(idir)
+                {
+                    if (iv_tmp[idir] > m_domain_box.bigEnd(idir))
+                    {
+                        iv_tmp[idir] = m_domain_box.bigEnd(idir);
+                    }
+                    else if (iv_tmp[idir] < m_domain_box.smallEnd(idir))
+                    {
+                        iv_tmp[idir] = m_domain_box.smallEnd(idir);
+                    }
+                }
+                value_at_point[i] = out_box(iv_tmp, icomp);
+                r_at_point[i] = Coordinates<double>::get_radius(
+                    iv_tmp, m_dx, {m_center[0], m_center[1], m_center[2]});
+            }
+        }
+
+        // assume some radial dependence and fit it
+        double analytic_change = 0.0;
+        // comp = const
+        if (order == 0)
+        {
+            analytic_change = 0.0;
+        }
+        // comp = B + A*r
+        else if (order == 1)
+        {
+            double delta_r_in_domain = r_at_point[1] - r_at_point[0];
```

Comment:
Note to self, this is written as if considering a point on the left hand side of the box. The value of r is therefore increasing to the left (a bit counterintuitively). If we consider the right hand side of the box, we would write:
    dr = r_0 - r_1
    A = (value_0 - value_1) / dr
which would give exactly the same value for A since the minus signs cancel, but would then make more sense when looking at `double delta_r_here = radius - r_at_point[0];`